pkg/controller/master/upgrade/upgrade_controller.go

```diff
@@ -272,6 +281,21 @@ func (h *upgradeHandler) OnChanged(key string, upgrade *harvesterv1.Upgrade) (*h
 				return h.upgradeClient.Update(toUpdate)
 			}
 		} else {
+			// save the original value of replica-replenishment-wait-interval setting and extend it with a longer value
+			// skip if the value is already larger than extendedReplicaReplenishmentWaitInterval
+			replicaReplenishmentWaitIntervalValue, err := h.getReplicaReplenishmentValue()
+			if err != nil {
+				return nil, err
+			}
+			if replicaReplenishmentWaitIntervalValue < extendedReplicaReplenishmentWaitInterval {
+				if err := h.saveReplicaReplenishmentToUpgradeAnnotation(toUpdate); err != nil {
+					return nil, err
+				}
```

Comment:
Can we store the annotation in the LH setting instead? so we don't need to update two separated objects?

Reply:
The upgrade CR here is going to be updated with or without this function call, so there is no additional update overhead imposed here. We're just hitchhiking the original workflow.